Mathlib/Tactic/Mabel.lean

```diff
@@ -0,0 +1,94 @@
+/-
+Copyright (c) 2024 Sam Ezeh. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sam Ezeh
+-/
+import Mathlib.Tactic.NormNum
+import Mathlib.Util.AtomM
+import Mathlib.Tactic.Abel
+import Qq
+
+/-!
+# The `mabel` tactic
+
+Evaluate expressions in the language of multiplicative, commutative monoids and groups.
+
+-/
+
+set_option autoImplicit true
+
+namespace Mathlib.Tactic.Mabel
+open Lean Elab Meta Tactic Qq Expr PrettyPrinter
+
+initialize registerTraceClass `mabel
+initialize registerTraceClass `mabel.detail
+
+lemma Additive_identity : Additive x = x := by rfl
+lemma ofMul_identity (x : α) : Additive.ofMul x = x := by
+  unfold Additive.ofMul
+  rfl
+
+/-- Tactic for converting equations in the language of
+multiplicative, commutative monoids and groups to those in the language
+of additive, commutative monoids and groups.
+-/
+syntax (name := to_additive) "to_additive" : tactic
+
+/-- The `to_additive` tactic, which solves converts equations in the language of
+multiplicative commutative additive groups (or monoids) to those in additive structures. -/
+elab_rules : tactic | `(tactic| to_additive) => withMainContext do
+  let some (_, e₁, e₂) := (← whnfR <| ← getMainTarget).eq?
+    | throwError "to_additive requires an equality goal"
+  let lhs ← delab e₁
+  let rhs ← delab e₂
+  evalTactic $ ← `(tactic|
+    conv => {
+      lhs
+      rw [← ofMul_identity $lhs];
+    } <;>
+    conv => {
+      rhs
+      rw [ ← ofMul_identity $rhs]
+    } <;>
+    simp [
+      ofMul_mul, ofMul_pow, ofMul_one,
+      ofMul_inv, ofMul_div, ofMul_zpow,
+      ofMul_eq_zero, ofMul_div
+    ])
```

Comment:
I think this needs improvement. For one thing, going via `delab` seems wrong to me!

Regarding `to_additive at h`, I think the skeleton would look like this:

```lean
syntax (name := to_additive) "to_additive" (location)? : tactic

elab_rules : tactic | `(tactic| to_additive $[$loc]?) => withMainContext do
  let loc := (loc.map expandLocation).getD (.targets #[] true)
  withLocation loc
    (atLocal := fun h ↦ do
      pure ()
    )
    (atTarget := do
      pure ()
    )
    (failed := fun _ ↦ throwError "to_additive made no progress")

```